plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundStore.java

```diff
@@ -123,25 +167,144 @@ public boolean dropType(String typeName) {
         return false;
     }
 
+    /**
+     *
+     * There would be a "database contains" relationship between D and T, and
+     * there would be a "table contains" relationship between T and each
+     * attribute. The types of attributes of those nodes, and the fact that A2
+     * and A4 are partition keys would be tags of those nodes. The fact that the
+     * table T is in a particular file format (Parquet or Avro) would be a tag
+     * on the table node.
+     * 
+     */
     public void createTable(Table tbl) throws InvalidObjectException, MetaException {
-        // TODO Auto-generated method stub
+        openTransaction();
+        // HiveMetaStore above us checks if the table already exists, so we can
+        // blindly store it here
+        try {
+            Table tblCopy = tbl.deepCopy();
+            String dbName = tblCopy.getDbName();
+            String tableName = tblCopy.getTableName();
+            Map<String, Tag> tagsMap = new HashMap<>();
+            tblCopy.setDbName(HiveStringUtils.normalizeIdentifier(tblCopy.getDbName()));
+            tblCopy.setTableName(HiveStringUtils.normalizeIdentifier(tblCopy.getTableName()));
+            normalizeColumnNames(tblCopy);
+            Tag tblTag = createTag(tableName, tblCopy);
+            tagsMap.put(tbl.getTableName(), tblTag);
+            // create an edge to db which contains this table
+            EdgeVersionFactory evf = getGround().getEdgeVersionFactory();
+            EdgeFactory ef = getGround().getEdgeFactory();
+            String edgeId = dbName + "." + tableName;
+            Optional<Map<String, Tag>> tags = Optional.of(tagsMap);
+            Optional<Map<String, String>> parameters = Optional.of(tbl.getParameters());
+            // new node for this table
+            String nodeId = getGround().getNodeFactory().create(tableName).getId();
+            NodeVersion nvTbl = getGround().getNodeVersionFactory().create(tags, Optional.empty(), Optional.empty(),
+                    parameters, nodeId, Optional.empty());
+            Edge edge = ef.create(edgeId);
+            String dbNode = dbMap.get(dbName);
```

Comment:
not clear. What we get from the GroundReadWrite is a handle to NodeVersion for the table we just created and a new edge is created using dbNodeId (from) and tableId nvTbl.getId() (to)

Can you explain?